File: Erik_Vincent/Assignment1/Assignment1/Model/QLS/AST/Section.cs
using System.Collections.Generic;
using System.Linq;

namespace Assignment1.Model.QLS.AST
{
    public class Section : Statement
    {
        public string Label { get; }
        public IReadOnlyCollection<Statement> Contents => _contents.AsReadOnly();

        private readonly List<Statement> _contents;

        public Section(string label, IEnumerable<Statement> contents)
        {
            Label = label;
            _contents = contents.ToList();
        }

        public override void Accept(IQLSASTVisitor visitor) => visitor.Visit(this);
    }
}

File: Remi_Hasan/src/main/java/qls/model/statement/Section.java
package qls.model.statement;

import qls.visitor.IQLSVisitor;

import java.util.List;

public class Section extends Statement {
    private final String title;
    private final List<Statement> statements;

    public Section(String title, List<Statement> statement) {
        this.title = title;
        this.statements = statement;
    }

    public String getTitle() {
        return title;
    }

    public List<Statement> getStatements() {
        return statements;
    }

    @Override
    public <T> T accept(IQLSVisitor<T> visitor) {
        return visitor.visit(this);
    }
}
